src/libs/Network/SequentialQueue.js

```diff
@@ -44,7 +44,12 @@ function process() {
 
     // Set the current request to a promise awaiting its processing so that getCurrentRequest can be used to take some action after the current request has processed.
     currentRequest = Request.processWithMiddleware(requestToProcess, true)
-        .then(() => {
+        .then((responseData) => {
+            // While processing the request, we might return the property to pause the queue if we notice that we're out of date. We're doing this here so we avoid
```

Comment:
I think the circular dependency doesn't really need mentioned in the comment, and it could do better to explain the purpose of this. Something like:

> A response might indicate that the queue should be paused. This happens when a gap in onyx updates are detected between the client and the server and that gap needs resolved before the queue can commence.

Should this also have a return or is it OK to run the next couple of lines even though the queue is paused?

Reply:
It is ok to finish, if we don't finish the others we'll break the app.